File: src/main/java/com/gclasscn/base/util/HostUtil.java
package com.gclasscn.base.util;

import java.net.MalformedURLException;
import java.net.URL;

import org.apache.http.HttpHost;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class HostUtil {

	private static final Logger logger = LoggerFactory.getLogger(HostUtil.class);
	
	public static HttpHost[] hosts(String[] urls){
		HttpHost[] hosts = new HttpHost[urls.length];
		for(int i=0; i<urls.length; i++){
			URL url = null;
			try {
				url = new URL(urls[i]);
			} catch (MalformedURLException e) {
				logger.error("URL协议、格式或者路径错误: {}", e.getMessage(), e);
				continue;
			}
			hosts[i] = new HttpHost(url.getHost(), url.getPort(), url.getProtocol());
		}
		return hosts;
	}
}

File: src/main/java/com/gclasscn/base/util/HostUtil.kt
package com.gclasscn.base.util

import org.slf4j.LoggerFactory
import org.apache.http.HttpHost
import java.net.URL
import java.net.MalformedURLException

object HostUtil {
	
	private var logger = LoggerFactory.getLogger(HostUtil::class.java);
	
	fun hosts(urls: Array<String>): ArrayList<HttpHost>{
		var hosts = ArrayList<HttpHost>(urls.size)
		urls.forEach {
			str -> var url: URL = URL(str)
			hosts.add(HttpHost(url.host, url.port, url.protocol))
		}
		return hosts
	}
}